feat: add channel gating

Complexlity opened the pull request.

Complexlity committed:
feat: add channel gating

vikiival commented on src/routes/gallery.tsx:
```diff
@@ -21,33 +21,118 @@ export const app = new Frog<HonoEnv>({})
-  const action = `/${chain}/${id}/finish`
+  // app.frame('/', async (c) => {
+
+  const { chain, id } = c.req.param();
+  //MOCK
+  // const chain = 'ahp'
+  // const id = '106'
```
```suggestion
  const { chain, id } = {
     chain: 'base',
     id: '0xd9a2c93ba2e9fae10fe762a42ee807bbf95764cc',
   } //c.req.param()
```

Complexlity replied:
The entire initial code actually remains unchanged. So it would work exactly the same as before by going to /base/${id}. The commented mocks were only used to make the new frame page and removed after that.

vikiival replied:
> The entire initial code actually remains unchanged

Isnt it better to make a new route than make a lot of conditionals?

Complexlity replied:
My Idea was, since I wasn't able to test, to reduce how much I refactor things. But that's cleaner for sure. Should be straight forward to refactor. On it

vikiival replied:
> I wasn't able to test


`0xd9a2c93ba2e9fae10fe762a42ee807bbf95764cc` this collection is mintable for  ` 0.0015 ETH` which can be reimbursed

Complexlity replied:
Alright. Will try it. For some reason, dyndata works now.

Done: https://basescan.org/tx/0x5999769168b7793dc1166762885188e2dab6e5e572fd2d6b8a249495a299ee3e. Nice

Complexlity committed:
chore: update chain, id params

vikiival commented on src/routes/gallery.tsx:
```diff
@@ -21,33 +21,118 @@ export const app = new Frog<HonoEnv>({})
 //   })
-// })  
+// })
+function UnverifiedImage({ imageUrl }: {imageUrl: string}) {
```
![Screenshot 2024-04-30 at 10 15 34](https://github.com/vikiival/based-frog-frame/assets/22471030/acafc497-b941-4d17-99a4-44f77092e9cd)


This is not very nice but can be updated by us later 

cc @exezbcz

exezbcz replied:
yeah can be updated later, it is functional

Complexlity committed:
refactor: logic to new route

vikiival committed:
Merge remote-tracking branch 'origin/main' into feat-channel-gate

vikiival committed:
feat: Add getUserChannelStatus function to fetch user channel status

vikiival committed:
chore: Remove commented out code and unused imports in follow.tsx

vikiival committed:
feat: Enable follow route in app

The follow route was commented out in the code changes. This commit enables the follow route in the app.

vikiival committed:
new constant JOIN_IMAGE with the URL for the join image.

vikiival committed:
chore: Update wrangler.toml with new environment configuration

JustLuuuu commented:
artist asking if we can add also follow his profile on warpacst too: https://warpcast.com/wiw

vikiival commented:
Cc @exezbcz ?
